### PR TITLE
Reworked integration tests to use Arquillian Cube instead of OTA

### PR DESCRIPTION
**Pull request by spisiakm**

### CODE REVIEW

**Comment by spisiakm** [comment]
Don't merge until https://github.com/arquillian/arquillian-cube/issues/835 is resolved. Thanks !

File: pom.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
     <vertx.version>3.4.2</vertx.version>
     <vertx-maven-plugin.version>1.0.9</vertx-maven-plugin.version>
     <vertx.verticle>io.openshift.booster.HttpApplication</vertx.verticle>
+    <version.arquillian_cube>1.9.0</version.arquillian_cube>
```
Comment (Ladicek) [inline pom.xml]: Why not 1.9.1?

File: pom.xml [hunk 2]
```diff
@@ -113,6 +121,31 @@
...
+        <exclusion>
+          <artifactId>undertow-core</artifactId>
+          <groupId>io.undertow</groupId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+
+    <dependency>
+      <groupId>org.arquillian.cube</groupId>
+      <artifactId>arquillian-cube-requirement</artifactId>
```
Comment (Ladicek) [inline pom.xml]: Do you actually use this?

File: src/test/resources/arquillian.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns="http://jboss.org/schema/arquillian"
+            xsi:schemaLocation="http://jboss.org/schema/arquillian
+               http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
+
+  <extension qualifier="openshift">
+    <property name="namespace.use.current">false</property>
```
Comment (Ladicek) [inline src/test/resources/arquillian.xml]: `true` please (and `enableImageStreamDetection` to `false`)

File: src/test/java/io/openshift/booster/OpenShiftIT.java [hunk 4]
```diff
@@ -22,28 +24,24 @@
  * Check the behavior of the application when running in OpenShift.
  */
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(Arquillian.class)
 public class OpenShiftIT {
 
-    private static OpenShiftTestAssistant assistant = new OpenShiftTestAssistant();
+    private final String applicationName = "configmap-vertx";
```
Comment (Ladicek) [inline src/test/java/io/openshift/booster/OpenShiftIT.java]: This is wrong as we now know :-)

File: src/test/java/io/openshift/booster/OpenShiftIT.java [hunk 5]
```diff
@@ -22,28 +25,23 @@
  * Check the behavior of the application when running in OpenShift.
  */
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(Arquillian.class)
 public class OpenShiftIT {
 
-    private static OpenShiftTestAssistant assistant = new OpenShiftTestAssistant();
+    @AwaitRoute
+    @RouteURL("${app.name}")
```
Comment (Ladicek) [inline src/test/java/io/openshift/booster/OpenShiftIT.java:32]: I'd suggest putting `@AwaitRoute` _after_ `@RouteURL`, but it's your choice :-)

**Comment by Ladicek** [comment]
The problem with deploying the role binding isn't fixed in Arquillian Cube yet, this shouldn't have been merged.

**Comment by cescoffier** [comment]
Just realizing this :-(

**Pull request merged**